src/cli2/instanceConfig.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+
+import {CliPlugin} from "./cliPlugin";
+import {bundledPlugins as getAllBundledPlugins} from "./bundledPlugins";
+
+type PluginName = string;
+
+export type InstanceConfig = {|
+  +bundledPlugins: Map<PluginName, CliPlugin>,
+|};
+
+export type RawInstanceConfig = {|
+  +bundledPlugins: $ReadOnlyArray<BundledPluginSpec>,
+|};
+
+// Plugin identifier, like `sourcecred/identity`. Version number is
+// implicit from the SourceCred version. This is a stopgap until we have
+// a plugin system that admits external, dynamically loaded
+// dependencies.
+export type BundledPluginSpec = string;
+
+type JsonObject =
+  | string
+  | number
+  | boolean
+  | null
+  | JsonObject[]
+  | {[string]: JsonObject};
```

Comment:
We (@decentralion and I) chatted about this a bit. We chose to write the
correct-style parser here (perhaps because I was driving ;-) ) because
it really does give us much stronger guarantees than the `any`-downcast,
though we definitely aren’t big fans of the verbosity. I’m not super
inclined to reflect Flow types into parsers, but I’m up for abstracting
this with an attoparsec-like or optics-based approach if we think that
that would be helpful. But later, not now; let’s find out how we
actually use these first.